File: solutions/julia/raindrops/raindrops.jl
function raindrops(number)
    sounds = ""

    if number % 3 == 0
        sounds *= "Pling"
    end

    if number % 5 == 0
        sounds *= "Plang"
    end

    if number % 7 == 0
        sounds *= "Plong"
    end

    if sounds == ""
        sounds = string(number)
    end

    return sounds
end

File: solutions/haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

import Data.List as List

convert :: Int -> String
convert n
    | sound == "" = show n
    | otherwise = sound
    where
        sound =
            List.intercalate "" [
                if n `mod` 3 /= 0 then "" else "Pling",
                if n `mod` 5 /= 0 then "" else "Plang",
                if n `mod` 7 /= 0 then "" else "Plong"
            ]

